sim7000-async/src/modem/mod.rs

```diff
@@ -198,28 +198,31 @@ impl<'c, P: ModemPower> Modem<'c, P> {
             dte_by_dce: FlowControl::Hardware,
         };
 
-        let commands = self.commands.lock().await;
-
-        for _ in 0..5 {
-            if let Ok(Ok(_)) = with_timeout(Duration::from_millis(2000), async {
-                commands.run(set_flow_control).await
-            })
-            .await
-            {
-                break;
+        // block for commands to avoid mutability problems
+        {
+            let commands = self.commands.lock().await;
+
+            for _ in 0..5 {
+                if let Ok(Ok(_)) = with_timeout(Duration::from_millis(2000), async {
+                    commands.run(set_flow_control).await
+                })
+                .await
+                {
+                    break;
+                }
             }
-        }
-        commands.run(ate::SetEcho(false)).await?;
-        commands
-            .run(cgreg::ConfigureRegistrationUrc::EnableRegLocation)
-            .await?;
+            commands.run(ate::SetEcho(false)).await?;
+            commands
+                .run(cgreg::ConfigureRegistrationUrc::EnableRegLocation)
+                .await?;
 
-        self.wait_for_registration(&commands).await?;
+            self.wait_for_registration(&commands).await?;
 
-        commands.run(cipmux::EnableMultiIpConnection(true)).await?;
-        commands.run(cipshut::ShutConnections).await?;
+            commands.run(cipmux::EnableMultiIpConnection(true)).await?;
+            commands.run(cipshut::ShutConnections).await?;
+        };
 
-        self.authenticate(&commands).await?;
```

Comment:
We _should_ be holding the lock for the entire function since we don't want anyone else to start sending AT-commands while the modem is activating. If lifetimes give you trouble, just move the body of `authenticate` in here.